Enterprise-CMCS/macpro-mdct-mfp
# MDCT-2878: SAR Creation

### Description


https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/19439679/54fd6764-9f85-4488-94cf-7baf197f3293

**What We Did:**
- Updated the SAR dashboard enough so that when you create a SAR report it shows in the table, or when visiting the page it shows any SARS in the DB
- Updated the SAR modal form so that it contains everything but the target populations question
- Allowed the ability to Edit the SAR on the dashboard
- Updated the SAR creation button so that there has to be an eligible work plan (Important Note: We are currently looking for Not Started and In Progress to copy from. We need to update this after Work Plan submission is completed/Dev work is closer to done to check for Work Plans that have been Submitted only) to copy from, otherwise it will show as disabled
- On SAR Creation, we return all the SARS metadata and form template like we do for MCPAR/MLR/WP, and we also include all of the associated Work Plans data in its own object in the fieldData. Going forward we can tweak it to better represent the fields we actually need in the SAR so it autopopulates the form

**What we didn't do:**

- Target populations is not in the modal. Will be a fast follow
- Create new tests. Will be a fast follow
- Reducing cognitive complexity. Will be a fast follow
- Worry about how the dashboard looks. Will be a separate ticket action.
- Worry about populating any fields in the SAR form itself. Will be in other tickets to expand upon this feature.

### Related ticket(s)


MDCT-2878

---
### How to test


- Navigate to SAR and see that there are none and you can't create one
- Create a Work plan. You can add some data to it if you like.
- Navigate to SAR page and see you can now add a submission
- Clicking create shows a modal with the last work plan you created as well as 1 choice list option
- Clicking create adds a new work plan in
- Checking the network tab should show in the preview pane that the work plan data you entered is there under fieldData
- Opening the SAR lets you see all of the SAR work!
- Back on the dashboard you can edit your SAR's choicelist option.

---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
